@xen-orchestra/backups/Backup.js

```diff
@@ -196,34 +196,47 @@ exports.Backup = class Backup {
       ...settings[schedule.id],
     }
 
+    const vmIds = extractIdsFromSimplePattern(job.vms)
     await using(
+      Disposable.all(
+        vmIds.map(id =>
+          this._getRecord('VM', id).catch(error => {
+            runTask(
+              {
+                name: 'get VM record',
+                data: { type: 'VM', id },
+              },
+              () => Promise.reject(error)
+            )
+          })
+        )
+      ),
```

Comment:
Does it work fine if you fetch the VM directly in `handleVm` and let its task handle the error?

Reply:
I don't see any issue that can be triggered by this behavior.

Reply:
Note: It will change the current behavior, are you OK with that?

Reply:
As long as it does not appear to break anything, yes.